core/src/main/java/org/apache/iceberg/FileMetadata.java

```diff
@@ -255,6 +255,8 @@ public DeleteFile build() {
       if (format == FileFormat.PUFFIN) {
         Preconditions.checkArgument(contentOffset != null, "Content offset is required for DV");
         Preconditions.checkArgument(contentSizeInBytes != null, "Content size is required for DV");
+        Preconditions.checkArgument(
+            referencedDataFile != null, "Referenced data file is required for DV");
```

Comment:
+1, could you please add a small test for this?

Reply:
Thanks for finding and fixing this @ebyhr! +1 for adding a simple test

Reply:
Thank you for your reviews! Added a test to TestDeleteFiles. Let me know if there is a better place.